sklearn/utils/estimator_checks.py

```diff
@@ -1190,6 +1189,33 @@ def check_estimators_pickle(name, estimator_orig):
         unpickled_result = getattr(unpickled_estimator, method)(X)
         assert_allclose_dense_sparse(result[method], unpickled_result)
 
+    # check if input with nans pickles properly
+    if name in ALLOW_NAN:
+        # set random 10 elements to be np.nan
+        rng = np.random.RandomState(42)
+        mask = rng.choice(X.size, 10, replace=True)
```

Comment:
maybe `replace=False` as it doesn't make sense to assign the same index twice to NaN..